packages/rancher-kiali-server/generated-changes/patch/templates/_helpers.tpl.patch

```diff
@@ -1,5 +1,14 @@
 --- charts-original/templates/_helpers.tpl
 +++ charts/templates/_helpers.tpl
+@@ -51,7 +51,7 @@
+ */}}
+ {{- define "kiali-server.selectorLabels" -}}
+ app.kubernetes.io/name: kiali
+-app.kubernetes.io/instance: {{ include "kiali-server.fullname" . }}
++app.kubernetes.io/instance: {{ .Release.Name }}
```

Comment:
Not sure if I agree with this change. We should be using the newer label on fresh installs; we should only preserve the older label if we can detect that the deployment already exists without the right label.

This can be done by checking if the [`.Release.IsUpgrade`](https://helm.sh/docs/chart_template_guide/builtin_objects/) and doing a [Helm `lookup`](https://helm.sh/docs/chart_template_guide/function_list/#lookup).

The result of the lookup should give you access to the fields of the k8s resource so you can pull the label that should be placed here from there. Alternatively, maybe if you just omit the label in this case it will preserve the existing one? Not sure though.

Reply:
Side note: https://helm.sh/docs/chart_best_practices/labels/ it does look like the best practice for app.kubernetes.io/instance label is to use the {{ .Release.Name}}

Reply:
@aiyengar2 I updated this per your recommendation and I agree with your reasonings for why I shouldn't just revert back to using {{ .Release.Name}}. My new code now checks if its .Release.IsUpgrade and then I grab the labels from the existing deployment and get the value for the `/instance` label that has been impacted. If the value matches the release name, I use that, if it doesn't then I default to the new label value (which is kiali normally). and then on fresh install its using the new label value.